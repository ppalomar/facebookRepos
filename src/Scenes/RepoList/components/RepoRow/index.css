.repo-row__container{
    display: flex;
    width: 270px;
    justify-content: space-between;
    color: #2196F3;
}

.repo-row__container:hover{
    background-color: #2196F3;
    color: #fff;
    cursor: pointer;
}

.repo-row__container .repo-name{
     padding: 10px;
}