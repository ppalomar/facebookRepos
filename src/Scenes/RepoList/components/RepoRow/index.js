import React from 'react';

import './index.css';

const RepoRow = ({ repoName, goDetail }) => {
  return (
    <div className="repo-row__container" onClick={() => goDetail(repoName)}>
      <span className="repo-name">{repoName}</span>
    </div>
  );
};

export default RepoRow;
