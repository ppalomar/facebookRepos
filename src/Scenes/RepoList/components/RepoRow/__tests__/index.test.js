import React from 'react';
import { shallow } from 'enzyme';

import RepoRow from '../index';

describe('RepoRow', () => {
  it('renders properly with basic props', () => {
    const props = {
      repoName: 'pablo_test',
      goDetail: () => {},
    };

    const wrapper = shallow(<RepoRow {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('when click on repo should trigger onClick', () => {
    const props = {
      repoName: 'pablo_test',
      goDetail: jest.fn(),
    };

    const root = shallow(<RepoRow {...props} />);
    root.find('.repo-row__container').simulate('click');
    expect(props.goDetail).toHaveBeenCalledWith(props.repoName);
  });
});
