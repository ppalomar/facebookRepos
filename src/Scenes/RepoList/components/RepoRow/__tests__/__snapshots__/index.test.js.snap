// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoRow renders properly with basic props 1`] = `
<div
  className="repo-row__container"
  onClick={[Function]}
>
  <span
    className="repo-name"
  >
    pablo_test
  </span>
</div>
`;
